EvtWeightTools/src/Data_to_MC_CorrectionInterface_0l_4tau_trigger.cc

```diff
@@ -40,132 +40,30 @@ Data_to_MC_CorrectionInterface_0l_4tau_trigger::getSF_triggerEff(TriggerSFsys ce
     sys = "down";
   }
 
-  double eff_2tau_tauLeg1_data = 0.;
-  double eff_2tau_tauLeg1_mc   = 0.;
-  double eff_2tau_tauLeg2_data = 0.;
-  double eff_2tau_tauLeg2_mc   = 0.;
-  double eff_2tau_tauLeg3_data = 0.;
-  double eff_2tau_tauLeg3_mc   = 0.;
-  double eff_2tau_tauLeg4_data = 0.;
-  double eff_2tau_tauLeg4_mc   = 0.;
-
-  if(std::fabs(hadTau1_eta_) <= 2.1 && aux::hasDecayMode(allowedDecayModes_, hadTau1_decayMode_))
-  {
-    eff_2tau_tauLeg1_data = tau_leg_efficiency(hadTau1_pt_, hadTau1_decayMode_, "ditau", wp_str_, "eff_data", sys);
-    eff_2tau_tauLeg1_mc   = tau_leg_efficiency(hadTau1_pt_, hadTau1_decayMode_, "ditau", wp_str_, "eff_mc", sys);
-  }
-
-  if(std::fabs(hadTau2_eta_) <= 2.1 && aux::hasDecayMode(allowedDecayModes_, hadTau2_decayMode_))
-  {
-    eff_2tau_tauLeg2_data = tau_leg_efficiency(hadTau2_pt_, hadTau2_decayMode_, "ditau", wp_str_, "eff_data", sys);
-    eff_2tau_tauLeg2_mc   = tau_leg_efficiency(hadTau2_pt_, hadTau2_decayMode_, "ditau", wp_str_, "eff_mc", sys);
-  }
-
-  if(std::fabs(hadTau3_eta_) <= 2.1 && aux::hasDecayMode(allowedDecayModes_, hadTau3_decayMode_))
-  {
-    eff_2tau_tauLeg3_data = tau_leg_efficiency(hadTau3_pt_, hadTau2_decayMode_, "ditau", wp_str_, "eff_data", sys);
-    eff_2tau_tauLeg3_mc   = tau_leg_efficiency(hadTau3_pt_, hadTau3_decayMode_, "ditau", wp_str_, "eff_mc", sys);
-  }
+  std::vector<const RecoHadTau*> matched_hadTaus;
 
-  if(std::fabs(hadTau4_eta_) <= 2.1 && aux::hasDecayMode(allowedDecayModes_, hadTau4_decayMode_))
+  for (auto hadTau_recObj : hadTau_recObjs)
   {
-    eff_2tau_tauLeg4_data = tau_leg_efficiency(hadTau4_pt_, hadTau4_decayMode_, "ditau", wp_str_, "eff_data", sys);
-    eff_2tau_tauLeg4_mc   = tau_leg_efficiency(hadTau4_pt_, hadTau4_decayMode_, "ditau", wp_str_, "eff_mc", sys);
+    //can we use 64, 128 filterbits only as they are related to di-tau trigger
+    //https://cmssdt.cern.ch/lxr/source/PhysicsTools/NanoAOD/python/triggerObjects_cff.py?v=CMSSW_12_6_X_2022-09-08-2300 #Line-0122
+    if ( hadTau_recObj->filterBits() ) matched_hadTaus.push_back(hadTau_recObj);
   }
 
-  double prob_data = 0.;
-  double prob_mc   = 0.;
-
-  for(int tau1_status = k2tau; tau1_status <= kNot2tau; ++tau1_status)
+  double prob_data(1);
+  double prob_mc(1);
```

Comment:
Why this syntax? Please keep the assignment notation since people are more familiar with it.